repository.go

```diff
@@ -77,35 +77,70 @@ type Repository struct {
 	wt billy.Filesystem
 }
 
-type InitOptions struct {
-	// The default branch (e.g. "refs/heads/master")
-	DefaultBranch plumbing.ReferenceName
+type initOptions struct {
+	defaultBranch plumbing.ReferenceName
+	workTree      billy.Filesystem
+	objectFormat  formatcfg.ObjectFormat
+}
+
+func newInitOptions() initOptions {
+	return initOptions{
+		defaultBranch: plumbing.Master,
+		workTree:      nil,
+		objectFormat:  "",
+	}
+}
+
+type InitOption func(*initOptions)
+
+// WithDefaultBranch sets the default branch for the new repo (e.g. "refs/heads/master").
+func WithDefaultBranch(b plumbing.ReferenceName) InitOption {
+	return func(o *initOptions) {
+		o.defaultBranch = b
+	}
+}
+
+// WithWorkTree sets the internal worktree for the new repo.
+func WithWorkTree(worktree billy.Filesystem) InitOption {
+	return func(o *initOptions) {
+		o.workTree = worktree
+	}
+}
+
+// WithBare is for backwards compatibility and is the opposite of WithWorkTree; a bare repo sets the worktree to nil.
+func WithBare() InitOption {
+	return func(o *initOptions) {
+		o.workTree = nil
+	}
+}
+
```

Comment:
I understand the idea behind having something for users to relate to. But it feels largely redundant.

```suggestion
```